### PR TITLE
add documentation on running unit tests

### PR DESCRIPTION
**Pull request by boegel**
rendered version available at http://boegel-eb.readthedocs.org/en/unit_test_docs/Unit-tests.html

### CODE REVIEW

File: docs/Unit-tests.rst [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+Running the full EasyBuild framework unit test suite takes about 15-40 minutes, depending on your system resoures and
+testing configuration (see for example https://jenkins1.ugent.be/view/EasyBuild/).
+
+easyblocks unit tests
+~~~~~~~~~~~~~~~~~~~~~
+
+The easyblocks unit test suite consists of a coupe of light-weight tests that are run per easyblock:
+
+* initlialising the easyblock, to check for Python syntax errors and faulty imports
```
Comment (fgeorgatos) [inline docs/Unit-tests.rst]: init...what?

File: docs/Unit-tests.rst [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+large-scale overall tests.
+
+For each of the available easyconfig files, the following aspects are tested:
+
+* parsing the easyconfig file, to make sure no syntax errors are present
+* verifying that the filename matches the contents of the easyconfig file (software name/version, version suffix and
+  toolchain name/version)
+* creating an ``EasyBlock`` instance using the parsed easyconfig, to check whether mandatory easyconfig parameters
+  are defined
+* ensuring that all requires patch files are available (right next to the easyconfig file)
```
Comment (fgeorgatos) [inline docs/Unit-tests.rst]: required

File: docs/Unit-tests.rst [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+* each pull request (update), see https://jenkins1.ugent.be/view/pull-request-builder/
+* each (set of) change(s) that is merged in (usually via a pull request)
+
+The status of the different unit test suites is tracked separately for the ``master`` and ``develop`` branches
+of the EasyBuild code repositories:
+
+* ``master`` (stable, latest EasyBuild release): https://jenkins1.ugent.be/view/EasyBuild/
+* ``develop`` (development, potentially unstable): https://jenkins1.ugent.be/view/EasyBuild%20(develop)/
+
+We strictly adher to the policy of only merging pull requests for which the correspondig (latest) run of the *full* unit
```
Comment (fgeorgatos) [inline docs/Unit-tests.rst]: ..nding

**Comment by fgeorgatos** [comment]
ok!

**Comment by boegel** [comment]
Thanks for the review @fgeorgatos!

**Comment by pforai** [comment]
Good stuffs! Was long due, but finally here :+1:

**Pull request merged**